src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -535,21 +520,36 @@ impl Renderer for HtmlHandlebars {
         fs::create_dir_all(destination)
             .with_context(|| "Unexpected error when constructing destination path")?;
 
-        let mut is_index = true;
+        let mut seen_index = false;
+        let mut first_non_draft = None;
         for item in book.iter() {
             let ctx = RenderItemContext {
                 handlebars: &handlebars,
                 destination: destination.to_path_buf(),
                 data: data.clone(),
-                is_index,
+                is_first_chapter: first_non_draft.is_none(),
                 book_config: book_config.clone(),
                 html_config: html_config.clone(),
                 edition: ctx.config.rust.edition,
                 chapter_titles: &ctx.chapter_titles,
             };
-            self.render_item(item, ctx, &mut print_content)?;
-            // Only the first non-draft chapter item should be treated as the "index"
-            is_index &= !matches!(item, BookItem::Chapter(ch) if !ch.is_draft_chapter());
+            if let Some(rendered_file) = self.render_item(item, ctx, &mut print_content)? {
+                seen_index |= rendered_file == Path::new("index.html");
+                if first_non_draft.is_none() {
+                    first_non_draft = Some(rendered_file);
+                }
+            }
+        }
+        if !seen_index {
+            if let Some(first_file) = first_non_draft {
+                let redirect_to = first_file
+                    .to_str()
```

Comment:
Changing PR to draft because this implementation is buggy and shouldn't be merged. Simply converting path to string can produce bad URLs.